### ISSUE TITLE
Error: Invalid uri, unknown variant `always_online`

### ISSUE THREAD

**Issue by rs-garrick** [cloudflare/terraform-provider-cloudflare#3199]
### Confirmation

- [X] My issue isn't already found on the issue tracker.
- [X] I have replicated my issue using the latest version of the provider and it is still present.

### Terraform and Cloudflare provider version

Terraform v1.6.5
on linux_amd64
+ provider registry.terraform.io/cloudflare/cloudflare v4.26.0
+ provider registry.terraform.io/hashicorp/http v3.4.2

### Affected resource(s)

cloudflare_custom_pages

### Terraform configuration files

```hcl
I removed this from my module:

resource "cloudflare_custom_pages" "terraform_managed_resource_always_online" {
  state   = "customized"
  type    = "always_online"
  url     = "${var.custom_pages_base}/always-on.html"
  zone_id = cloudflare_zone.terraform_managed_resource_zone.id
}
```

### Link to debug output

https://gist.github.com/rs-garrick/a73b97898ee786a52dcb9c7d7f1c4fc2

### Expected output

I expected Terraform plan to delete the `always_online` custom page configuration.

### Actual output

This is repeated for every zone in my configuration:
```
│ Error: Invalid uri, unknown variant `always_online`, expected one of `basic_challenge`, `waf_challenge`, `waf_block`, `ratelimit_block`, `country_challenge`, `managed_challenge`, `ip_block`, `under_attack`, `500_errors`, `1000_errors` (1211)
```

### Steps to reproduce

Not sure this can be reproduced any more.  After `always_online` was removed from Cloudflare, Terraform can no longer remove them. I'm not sure how to move forward.

### Additional factoids

You can clearly see in the debug output, TF calls `/custom_pages/always_online`. This endpoint errors now. This breaks planning. So anyone with such a custom page in their TF state is now stuck.

### References

https://github.com/cloudflare/terraform-provider-cloudflare/pull/3117

**Comment by Nmishin** [comment]
this is expected, please check "BREAKING CHANGES" for 4.25.0 version of provider:
https://github.com/cloudflare/terraform-provider-cloudflare/blob/master/CHANGELOG.md#4250-february-21st-2024

**Comment by rs-garrick** [comment]
I'm not going to upload more debug output. It's not needed. The Cloudflare provider just broke state for a bunch of people.

I'm just leaving this as a note to future users who get stuck.  The only way forward is to manually remove those custom pages from the state.

I was able to do `terraform state rm $(terraform state list | grep always_online)` but your YMMV depending on what you named your resources.

**Comment by deniojunior-hotmart** [comment]
It seems related to a breaking change on version [v4.25.0](https://github.com/cloudflare/terraform-provider-cloudflare/releases/tag/v4.25.0).

> BREAKING CHANGES:
> resource/cloudflare_custom_pages: Removed the always_online variant. This page is never generated anymore, if a requested page is unavailable in the archive the error page that would have been shown if always online wasn't enabled is shown. (https://github.com/cloudflare/terraform-provider-cloudflare/issues/3117)

**Comment by rs-garrick** [comment]
@deniojunior-hotmart Yes, I linked that to that change in my initial comment, but it was NOT "fixed" in v4.25.0.  Support for `always_online` was removed which left people with a broken state.  Anyone that had an `always_online` page in their state, is left unable to create a terraform plan.  The resource must be manually removed before Terraform will function again.
